### PR TITLE
Prompt tokenization bugfix

### PR DESCRIPTION
**Pull request by vadam5**
# What does this PR do ?

1. adds <> around virtual prompt token string
2. Removes pseudo token configurability from the config file in favor of using a consistent pseudo token set by us
3. Makes PROMPT lowercase

**Collection**: BigNLP

# Changelog 
Updated megatron_gpt_prompt_learning_model
Updated prompt learning dataset unit tests
Added Enum for virtual prompt token string

# Usage
Same as before

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
- [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: tests/collections/nlp/test_prompt_learning.py [hunk 1]
```diff
@@ -71,8 +71,8 @@ def get_task_templates():
         "task_id_num": 0,
     }
     task_templates['task name B'] = {
-        "prompt_template": "<|VIRTUAL_PROMPT_0|>{question}<|VIRTUAL_PROMPT_1|>{answer}{extra}",
-        "prompt_template_fields": ['question', 'answer'],
+        "prompt_template": "<|VIRTUAL_PROMPT_0|> {question} <|VIRTUAL_PROMPT_1|> {answer}{extra}",
```
Comment (yidong72) [inline tests/collections/nlp/test_prompt_learning.py:74]: why you add the space around the virtual tokens? If space is needed, virtual prompt should learn to be a space embedding?

File: tests/collections/nlp/test_prompt_learning.py [hunk 2]
```diff
@@ -83,6 +83,15 @@ def get_task_templates():
...
 
 
+def get_pseudo_tokens(total_virtual_tokens):
+    pseudo_tokens = [
+        VirtualPromptPlaceholderToken.BASE.value + str(i) + VirtualPromptPlaceholderToken.END.value
+        for i in range(total_virtual_tokens)
+    ]
+
+    return pseudo_tokens
+
```
Comment (yidong72) [inline tests/collections/nlp/test_prompt_learning.py:93]: I see you have the similar logics in the prompt_lenaring_model file. Put the function there?

**Review by yidong72 — COMMENTED** [review]
Only a tiny comment. Good job using the enum type.

File: tests/collections/nlp/test_prompt_learning.py (hunk 1 above)
Comment (vadam5) [reply tests/collections/nlp/test_prompt_learning.py:74]: I was playing around with the unit tests to help diagnose a bug with the huggingface tokenizer. This change isn't needed, I just forgot to put it back. (Turns out there is a bug with the huggingface tokenizer, btw)

File: tests/collections/nlp/test_prompt_learning.py (hunk 2 above)
Comment (vadam5) [reply tests/collections/nlp/test_prompt_learning.py:93]: Sure, I can do that

**Pull request merged**